lib/Service/FaceClusterAnalyzer.php

```diff
@@ -96,9 +100,22 @@ public function calculateClusters(string $userId) {
 				 * @var FaceCluster $cluster
 				 */
 				$cluster = $this->faceClusters->insert($cluster);
+
+				$clusterCentroid = self::calculateCentroidOfDetections($clusterDetections);
 			}
 			foreach ($notYetClustered as $item) {
+				/** @var FaceDetection $detection */
 				$detection = $item[0];
+				$distance = new Euclidean();
+				if ($detection->getThreshold() !== 0.0) {
```

Comment:
```suggestion
				if ($detection->getThreshold() > 0.0) {
```

Maybe it dates back from C practice but unequal comparison with float scares me.

Reply:
Good point! :)